### ISSUE TITLE
Launcher as a desktop.

### ISSUE THREAD

**Issue by me4oslav** [ozonos/atom-launcher#12]
The whole thing with an empty workspace in Gnome is such an outdated metaphor. To fix it what we can make is this - if the user gets to an empty workspace he doesn't get to stare at his wallpaper until he starts an app, but rather the app launcher. When he opens a window, the app launcher goes away. If he closes all the windows the app launcher appears again.
This also NEEDS an off switch, because people will kill us if we don't have one.
Can somebody please implement that, because starring at the wallpaper is counter productive. :P

**Comment by me4oslav** [comment]
Also, because Google+ shows a pic of my majestic hair as a thumbnail if I try to post it there and it is too damn gorgeous and will distract folks, here have the ozon logo picture.
![pic](https://cloud.githubusercontent.com/assets/1658870/6327720/e8cff7be-bb67-11e4-935a-63d4e1f3719e.png)
P.S. I wanted to add a picture of Katherine, but that would have been an overkill.

**Comment by zesterer** [comment]
I don't know about you... but I kind of like going onto a new workspace and looking at my desktop background. That's what it's there for.

**Comment by david-baiIey** [comment]
yes I agree with @zesterer tbh

**Comment by mpnordland** [comment]
Yes, hence the switch. I've started investigating how implement.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by cldx** [comment]
I think it's kinda weird, i use the overview to select a new workspace and it shoots me right back into the overview? 

If the Launchers would be uncoupled from the rest of the overview this would make more sense and also look nicer.

**Comment by me4oslav** [comment]
@zesterer @david-baiIey Hence like @mpnordland said, it will be a behaviour that can be disabled. I am well aware that this switch is needed. It is too much for a drastic change for some people. I for one see no point of staring at a wallpaper, instead of getting stuff done but some people do. Don't worry, you will be able to disable that.
@mpnordland YAY. And welcome back.
@cldx What do you mean by "If the Launchers would be uncoupled from the rest of the overview this would make more sense and also look nicer."? Kinda didn't get it.

**Comment by cldx** [comment]
Uncoupled, as in we can display only the Apps, without Workspaces and searchbar and without the dark background in this case.

**Comment by me4oslav** [comment]
@cldx Well, that's what I meant. :)
I said "Launcher as a desktop", not "Overview as a desktop". On empty desktop the user gets the app of grids, no workspaces and with subtle darkening. However why do you wanna have no search bar? It is super useful.

**Comment by cldx** [comment]
Oh okay, i guess we have the same thing in mind then :)

**Comment by david-baiIey** [comment]
@me4oslav I think you should do a G+ poll to see which people prefer, so that you know which one to use as **default**. It looks like it would be better for it to be opt-in rather than opt-out based on the comments here.

**Comment by me4oslav** [comment]
@david-baiIey Atom is very much in house designed desktop shell, we've never given community much power over Atom's design. A better option will be to have in enabled out for the box but make a super easy and prominent way to disable it.
@cldx It seems so. About the search - can we keep it? Oh and make a telegram account, so I can add you to Numix and Ozon chats there.

**Comment by smartfuse** [comment]
@david-baiIey agree with @me4oslav. This is our OS and my idea for launcher so we are keeping it :)

**Comment by me4oslav** [comment]
dat my idea argument, though. :D

**Comment by cldx** [comment]
@david-baiIey it's not even that we dislike community feedback, but its very clear what will come put of it, some will love it, some will hate it, the next guy wants an option to put it into the tray and so on.  

What i'd prefer is if we have branches for stuff like this and we can dogfood this to ourselves and just test it for a a while and then get back to it and see if it's really as awesome as we thought it would be or if it's one of those ideas that sound  good in theory but dont work in reality.

This is also a kind of democracy cause of course everybody can bug us on github and test the branches. Just not as widespread as G+

**Comment by me4oslav** [comment]
Imma gonna rename you to Jono Konopka. :)

**Comment by michealharker** [comment]
@cldx :+1: I like the idea of just having apps on the desktop, instead of all the workspaces stuff or so.

**Comment by mpnordland** [comment]
So, question: do we want to implement this in atom-launcher or atom-workspaces? Because we show the app grid when we switch to an empty workspace. Also, How subtle should the darkening be?

**Comment by me4oslav** [comment]
I guess it its for atom-launcher. Darkening should be 1/2 as little, I think. But we can tweak that.
BTW, @cldx tells me that @kyuucr might be able to help, also.

**Comment by kyuucr** [comment]
I could probably implement it when switching to new blank workspace, but triggering it when all windows are closed sounds like I need more digging through gnome shell source code. Anyhow I'll look into it.

**Comment by cldx** [comment]
@kyuucr i looked into this for a while now but ran into several problems, how about we team up on this and make a spec and figure out the blocking stuff together? I'll throw you a message tonight on G+ so we can start.

**Comment by kyuucr** [comment]
@cldx It's already late in here so hit me up and I'll read your message first time in the morning :D

**Comment by smartfuse** [comment]
Anyone want to actually fix this?
